### PR TITLE
feat(gate): noise-floor gate epsilon + paired significance stats

### PR DESCRIPTION
**Pull request by arthrod**
Stacked on #7 (renderer canary). Sixth PR in the robustness stack.

## What

- **`bench noise-floor --docs N`** double-renders N corpus docs through the full convert→rasterize→score path and records the measured render noise floor to `results/noise_floor.json` (n, sigma, max delta, per-doc deltas, LO version, dpi).
- **Gate epsilon becomes data-driven**: `gate()` gains an `eps` parameter and the CLI derives it as `max(1e-4, 3*sigma)` from the recorded file. No file → historical `1e-4` default, unchanged behavior.
- **Paired stats in RESULTS.md**: for script_redlines, pairwise win/loss/tie counts, median per-doc Δ, and a Wilcoxon signed-rank (zsplit) p-value over ≥20 shared docs — ranking gaps now come with significance attached.

## Measured floor (recorded, not assumed)

`bench noise-floor --docs 6` on the oracle machine: **n=6, sigma=0.000000, max=0.000000** — double renders are byte-identical, so the floor is exactly zero and the historical 1e-4 epsilon is already conservative. The module still earns its keep: a nonzero sigma in any environment is itself an alarm (rendering stopped being deterministic there), complementing the canary which only catches drift *between* environments.

## Tests

- 4 new tests in `tests/test_noise_floor.py` (stats, eps derivation, absent/corrupt file fallback, write/readback).
- 2 new tests in `tests/test_export_results_md.py` for the paired-stats section (agreeing pair suppressed below min-n, significance emitted above).
- Full suite: 262 passed, 2 skipped.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Make gate epsilon data-driven based on measured render noise floor and add paired significance statistics for script_redlines benchmark results.

New Features:
- Add a noise-floor benchmarking CLI command to measure render variability across double-rendered DOCX documents and persist results.
- Introduce a noise_floor module to derive gate epsilon from recorded render noise statistics and store measurement metadata.
- Extend export-results markdown to include paired per-document vendor comparisons with Wilcoxon-based significance metrics for script_redlines.

Enhancements:
- Allow the gate() function to accept an explicit epsilon threshold while preserving the historical default behavior.
- Wire gate execution to use epsilon derived from the noise-floor file when present.

Tests:
- Add unit tests for noise-floor statistics, epsilon derivation, and gate behavior with explicit and default thresholds.
- Add tests verifying inclusion and minimum-overlap behavior of the paired stats section in exported fidelity markdown.

### CODE REVIEW

**Comment by arthrod** [comment]
Closing: tip was already an ancestor of origin/main before #17 (landed earlier via the pure-juice merge path / intermediate stack merges). No unique commits remain to merge.

File: src/neurotic_docx_bench/cli.py [hunk 1]
```diff
@@ -1530,6 +1533,58 @@ def canary_cmd(
...
+        work = Path(work_s)
+        for i, docx in enumerate(docx_files):
+            sub = work / f"d{i}"
+            src_dir = sub / "src"
+            src_dir.mkdir(parents=True)
+            shutil.copy(docx, src_dir / docx.name)
+            r1 = renderer.to_pdfs(src_dir, sub / "r1", jobs=1)
+            r2 = renderer.to_pdfs(src_dir, sub / "r2", jobs=1, force=True)
+            p1 = next(iter([r.pdf for r in r1.results if r.ok and r.pdf]), None)
+            p2 = next(iter([r.pdf for r in r2.results if r.ok and r.pdf]), None)
```
[the remark this answers is not in the document]
Comment (arthrod) [reply src/neurotic_docx_bench/cli.py:1572]: Fixed in 5e1620d. `next((r.pdf for r in ...), None)` without intermediate list.

File: src/neurotic_docx_bench/noise_floor.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    """``max(DEFAULT_EPS, 3*sigma)`` from a recorded noise-floor file; the default
+    when the file is absent or unreadable."""
+    if not path.is_file():
+        return DEFAULT_EPS
+    try:
+        data = json.loads(path.read_text())
+        sigma = float(data.get("sigma", 0.0))
+    except (json.JSONDecodeError, OSError, TypeError, ValueError):
+        return DEFAULT_EPS
+    return max(DEFAULT_EPS, 3.0 * sigma)
```
[the remark this answers is not in the document]
Comment (arthrod) [reply src/neurotic_docx_bench/noise_floor.py:46]: Fixed in 5e1620d. Non-finite/negative sigma falls back to DEFAULT_EPS. Test: `test_eps_from_noise_floor_file`.
